### ISSUE TITLE
Experiment removing caching layer from `Lookup/UnorderedSet` to see if improvement

### ISSUE THREAD

**Issue by austinabell** [near/near-sdk-rs#731]
As these collections don't need the caching layer to have `std` like APIs and have the worst cost to benefit ratio of all collections, we should explore:

- [ ] Removing `StableMap` cache from `LookupSet`
- [ ] Removing `LookupMap` from `UnorderedSet`

And just doing the same operations with interacting with the store without caching changes/reads. This would need to be done before either of these is stabilized.

If anything is changed, should update anything relating to https://github.com/near/sdk-docs/issues/87 if that has been done already.

cc: @itegulov this is definitely low priority, but if anyone takes this on, you have the most context around this
